File: 82-remove-duplicates-from-sorted-list-ii/remove-duplicates-from-sorted-list-ii.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:

        dummy=ListNode()
        dummy.next=head

        prev=dummy
        crr=head
        while crr!=None and crr.next!=None:
            if crr.val==crr.next.val:
                while crr.next!=None and crr.val==crr.next.val:
                    crr=crr.next
                crr=crr.next
                prev.next=crr
            else:
                prev=crr
                crr=crr.next
        prev.next=crr

        return dummy.next

File: 82-remove-duplicates-from-sorted-list-ii/remove-duplicates-from-sorted-list-ii.java
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode dummy=new ListNode();
        dummy.next=head;
        
        ListNode prev=dummy,crr=head;

        while(crr!=null && crr.next!=null){
            if(crr.val==crr.next.val){
                while(crr.next!=null && crr.val==crr.next.val){
                    crr=crr.next;
                }
                prev.next=crr.next;
                crr=crr.next;
            }
            else{
                prev=crr;
                crr=crr.next;
            }
        }


        return dummy.next;
    }
}